webapp/packages/plugin-codemirror6/src/useEditorDefaultExtensions.ts

```diff
@@ -34,55 +35,90 @@ DEFAULT_KEY_MAP.push({
+  bracketMatching,
+  dropCursor,
+  highlightSelectionMatches,
+  crosshairCursor,
+  foldGutter: () =>
+    foldGutter({
+      markerDOM: (open: boolean) => {
+        const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+        svg.setAttributeNS(null, 'viewBox', '0 0 15 8');
+        svg.style.maxWidth = '100%';
+        svg.style.maxHeight = '100%';
 
-    extensions.push(
-      tooltips({
-        parent: document.body,
-      }),
-      highlightSpecialChars(),
-      highlightSelectionMatches(),
-      syntaxHighlighting(classHighlighter),
-      bracketMatching(),
-      dropCursor(),
-      crosshairCursor(),
-      foldGutter({
-        markerDOM: (open: boolean) => {
-          const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
-          svg.setAttributeNS(null, 'viewBox', '0 0 15 8');
-          svg.style.maxWidth = '100%';
-          svg.style.maxHeight = '100%';
+        const use = document.createElementNS('http://www.w3.org/2000/svg', 'use');
+        use.setAttributeNS('http://www.w3.org/1999/xlink', 'href', GlobalConstants.absoluteUrl('/icons/icons.svg#angle'));
+        svg.appendChild(use);
 
-          const use = document.createElementNS('http://www.w3.org/2000/svg', 'use');
-          use.setAttributeNS('http://www.w3.org/1999/xlink', 'href', GlobalConstants.absoluteUrl('/icons/icons.svg#angle'));
-          svg.appendChild(use);
+        const element = document.createElement('div');
+        element.appendChild(svg);
+        element.className = clsx('cm-gutterElement-icon', open ? 'cm-foldGutter-open' : 'cm-foldGutter-folded');
 
-          const element = document.createElement('div');
-          element.appendChild(svg);
-          element.className = clsx('cm-gutterElement-icon', open ? 'cm-foldGutter-open' : 'cm-foldGutter-folded');
+        return element;
+      },
+    }),
+  highlightActiveLineGutter,
+  highlightActiveLine,
+  indentOnInput,
+  rectangularSelection,
+  keymap: () => keymap.of(DEFAULT_KEY_MAP),
+  lineWrapping: () => EditorView.lineWrapping,
+};
 
-          return element;
-        },
-      }),
-      highlightActiveLineGutter(),
-      highlightActiveLine(),
-      indentOnInput(),
-      rectangularSelection(),
-      keymap.of(DEFAULT_KEY_MAP),
-    );
+const DEFAULT_EXTENSIONS_COMPARTMENT = new Compartment();
 
+/** Provides the necessary extensions to establish a basic editor */
+export function useEditorDefaultExtensions(options?: IDefaultExtensions): [Compartment, Extension] {
+  return useMemo(() => {
+    const extensions = Object.entries({ ...defaultExtensionsFlags, ...options } || {})
+      .filter(([, isEnabled]) => isEnabled)
+      .map(([key]) => {
+        const extensionFunction = extensionMap[key as keyof typeof extensionMap];
+        return extensionFunction?.();
+      })
+      .filter(Boolean);
     return [DEFAULT_EXTENSIONS_COMPARTMENT, extensions];
-  }, [options?.lineNumbers]);
+  }, [options]);
```

Comment:
`options` always will be a new object (so memo will not work at all), you can save previous options to the ref and then use  `isObjectsEqual` to determine changes